phpseclib/File/X509.php

```diff
@@ -1137,7 +1138,11 @@ public function validateSignature($caonly = true)
                                 $subjectKeyID = $this->getExtension('id-ce-subjectKeyIdentifier', $ca);
                                 switch (true) {
                                     case !is_array($authorityKey):
+                                    case !$subjectKeyID:
                                     case is_array($authorityKey) && isset($authorityKey['keyIdentifier']) && $authorityKey['keyIdentifier'] === $subjectKeyID:
+                                        if (isset($authorityKey['authorityCertSerialNumber']) && $authorityKey['authorityCertSerialNumber'] != $ca['tbsCertificate']['serialNumber']) {
```

Comment:
What if `$authorityKey` isn't an array (ie. it isn't set)? It'll then do the `isset($authorityKey['authorityCertSerialNumber'])`, trying to access the index of a null. It seems like this could produce a PHP error (E_NOTICE) but it actually doesn't appear to do so: https://3v4l.org/UFH9D 

```php
<?php
error_reporting(E_ALL);

$a = null;
echo isset($a['test']) ? 'yes' : 'no';
```
Be that as it may I still don't think it ought to basically be doing `isset(null['key'])`.

Reply:
It's just shortcut to avoid another condition in if statement widely used. But I don't mind to perform another check if will make it more readable